### PR TITLE
[skip ci] Support CentOS Storage SIG repository

### PR DESCRIPTION
**Pull request by mcodenie**
To declare this via all.yml, ie.

ceph_origin: **repository**
ceph_repository: **sig**

Note: Vagrant boxes got issues on the default scsi if using RHEL8/9 based need to change it to other types under the VirtualBox storage controller to get rid of issues in my test to work using virtio, [more info on this](https://access.redhat.com/solutions/3773611) . Alter line 522 in Vagrantfile, like the following.
          vb.customize ['storagectl', :id,
                        '--name', 'OSD Controller',
                        '--add', '**virtio**']
RHEL9-based distro got still issues few python modules with the CentOS SIG.

### CODE REVIEW

**Comment by mcodenie** [comment]
A further note on the Ceph version per SIG support currently:
Centos 7 (jewel, luminous, nautilus)
Centos 8 (octopus, pacific)
Centos 8 Stream/Rocky 8/Alma 8 (nautilus, octopus, pacific, quincy)
Centos 9 Stream (pacific, quincy) but still got issues on python3-jwt and python3-routes. Hope SIGs will fix this.

CentOS 6 and below not supported as per SIG repo.

**Review by guits — CHANGES_REQUESTED** [review]
could you please:

- squash your commits,
- add a sign off,
- add a decent commit message,

also, do not edit sample files directly, edit `./roles/ceph-defaults/defaults/main.yml` instead and run `./ceph-ansible/generate_group_vars_sample.sh`

thanks!

**Comment by mcodenie** [comment]
@guits done as per request. Can you please check if the commit message will suffice? Thanks.

**Comment by dsavineau** [comment]
Would it be better to use the existing ceph SIG packages instead of configuring the SIG repository manually ? (which wouldn't benefits from the CentOS mirrorlist)

```console
# dnf search centos-release-ceph
Last metadata expiration check: 0:01:03 ago on Wed Sep 28 13:05:57 2022.
===================================================================================================== Name Matched: centos-release-ceph ======================================================================================================
centos-release-ceph-nautilus.noarch : Ceph Nautilus packages from the CentOS Storage SIG repository
centos-release-ceph-octopus.noarch : Ceph Octopus packages from the CentOS Storage SIG repository
centos-release-ceph-pacific.noarch : Ceph Pacific packages from the CentOS Storage SIG repository
centos-release-ceph-quincy.noarch : Ceph Quincy packages from the CentOS Storage SIG repository
```

**Comment by mcodenie** [comment]
@dsavineau initially that's the plan, however, was changed when supposedly able to install even to the RHEL, and to AlmaLinux 9 which doesn't have SIGs support. Else will do that way then.

**Comment by guits** [comment]
> @dsavineau initially that's the plan, however, was changed when supposedly able to install even to the RHEL, and to AlmaLinux 9 which doesn't have SIGs support. Else will do that way then.

What's the initial goal of this PR? Adding `AlmaLinux` support? Or Ceph SIG repository support? Or both?

**Comment by mcodenie** [comment]
> > @dsavineau initially that's the plan, however, was changed when supposedly able to install even to the RHEL, and to AlmaLinux 9 which doesn't have SIGs support. Else will do that way then.
> 
> What's the initial goal of this PR? Adding `AlmaLinux` support? Or Ceph SIG repository support? Or both?
Maybe the focus is on Ceph SIG repository support via the binary packages as per suggested. Currently got issues on CentOS 8 upon testing (as the repos url moved to the vault) supposedly modified the code, and been doing something lately.

AlmaLinux perhaps will be on separate PR.

**Comment by mcodenie** [comment]
It seems the binary packages of SIG mirrors not working upon further testing on different distros, which still need to be manually altered the baseurl - same to this PR.

i.e on Rocky...
`Error: Failed to download metadata for repo 'centos-ceph-quincy': Cannot prepare internal mirrorlist: No URLs in mirrorlist
`

Default mirrorlist/baseurl of the SIG binary package.
```
cat /etc/yum.repos.d/CentOS-Ceph-Quincy.repo 
# CentOS-Ceph-Quincy.repo
...
[centos-ceph-quincy]
name=CentOS-$releasever - Ceph Quincy
mirrorlist=http://mirrorlist.centos.org/?release=$releasever&arch=$basearch&repo=storage-ceph-quincy
#baseurl=http://mirror.centos.org/$contentdir/$releasever/storage/$basearch/ceph-quincy/
gpgcheck=1
enabled=1
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-SIG-Storage

[centos-ceph-quincy-test]
.... 

[centos-ceph-quincy-source]
name=CentOS-$releasever - Ceph Quincy Source
baseurl=http://vault.centos.org/$contentdir/$releasever/storage/Source/ceph-quincy/
gpgcheck=1
enabled=0
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-SIG-Storage
```

Check the content directory variable.
```
cd /etc/dnf/vars;grep . *
contentdir:pub/rocky
sigcontentdir:pub/sig
```

Checking the version and OS architecture.
```
/usr/libexec/platform-python -c 'import dnf, json; db = dnf.dnf.Base(); print(json.dumps(db.conf.substitutions, indent=2))'
{
  "arch": "x86_64",
  "basearch": "x86_64",
  "releasever": "8"
}

```

In short, even the contentdir of Rocky does not exist on the base URL of CentOS. The mirrorlist only works if using the [8-stream](http://mirrorlist.centos.org/?release=8-stream&arch=x86_64&repo=storage-ceph-quincy) but not on [8 releasever](http://mirrorlist.centos.org/?release=8&arch=x86_64&repo=storage-ceph-quincy) as per above error. I still need to be manually altered though.

If you agree maybe a separate PR for SIG-binary-based packages as well, based on the above issues and unless that package (centos-release-ceph-<version>) has been corrected as only working well on 8-stream/9-stream.

**Comment by mcodenie** [comment]
> It seems the binary packages of SIG mirrors not working upon further testing on different distros, which still need to be manually altered the baseurl - same to this PR.
> 
> If you agree maybe a separate PR for SIG-binary-based packages as well, based on the above issues and unless that package (centos-release-ceph-) has been corrected as only working well on 8-stream/9-stream.

Disregard my previous comment (quoted here). Modified lately to use the SIG binary packages, however, modified the baseurl and mirrorlist to non 8-Stream. 

CentOS 8 not working and AlmaLinux. For 9 distros got issues on ceph-mgr-dashboard which got dependencies issues on python3-jwt and python3-routes.

In summary, only support Centos 8-Stream and Rocky 8 which works well.

File: roles/ceph-common/tasks/installs/centos_sig_repository.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    regexp:  "{{ item.regex }}"
+    replace: "{{ item.replace }}"
+  when:
+    - ansible_facts['os_family'] == 'RedHat'
+    - ansible_facts['distribution_major_version'] == "8"
+    - ansible_facts['distribution'] != 'CentOS' and ansible_facts['distribution_release'] != "Stream"
+  register: result
+  until: result is succeeded
+  with_items:
+    - {  regex:  '\?release\=\$releasever', replace: '?release=8-stream' }
```
Comment (mcodenie) [inline roles/ceph-common/tasks/installs/centos_sig_repository.yml:24]: original lines after the SIG installed...

**Comment by guits** [comment]
we probably need to add tests for this
